clippy_lints/src/doc/mod.rs

```diff
@@ -834,6 +836,10 @@ impl<'a, 'tcx> Visitor<'tcx> for FindPanicUnwrap<'a, 'tcx> {
             return;
         }
 
+        if let ExprKind::ConstBlock(_) = expr.kind {
+            self.is_const = true;
+        }
+
```

Comment:
If a `ConstBlock` exists within the initial expression, then it is possible to assume a constant environment.

Let me know if there is a better alternative.

Reply:
An alternative would be using [clippy_utils::is_const](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/fn.in_constant.html) before the visitor

Reply:
Exactly what I was looking for. Thanks!